AUTHORING_GUIDE.md

```diff
@@ -281,216 +191,203 @@ Args:
+* All tests should be independent of one another and order-independent.
 
-Again, descriptive names prevent you from having to exhaustively describe
-every parameter.
+## Arrange, Act, Assert
 
-Some samples demonstrate multiple functions. You should use *subparsers* to
-handle this, for example:
+Tests for samples should follow the “Arrange, Act, Assert” structure for
+unit tests:
 
-```python
-if __name__ == '__main__':
-    parser = argparse.ArgumentParser(
-        description=__doc__,
-        formatter_class=argparse.RawDescriptionHelpFormatter)
-    parser.add_argument('bucket_name', help='Your cloud storage bucket.')
+* _Arrange_ — create and configure the components required for the test.
+Avoid nesting; prioritize readability and simplicity over efficiency.
+* _Act_ — execute the code to be tested, such as sending a request and
+receiving a response.
+* _Assert_ — verify that the test results match what is expected.
 
-    subparsers = parser.add_subparsers(dest='command')
+## External Resources
 
-    subparsers.add_parser('list', help=list_blobs.__doc__)
+Whenever possible, tests should run against the live production version of
+cloud APIs and resources. This will assure that any breaking changes in those
+resources are identified by the tests.
 
-    upload_parser = subparsers.add_parser('upload', help=upload_blob.__doc__)
-    upload_parser.add_argument('source_file_name')
-    upload_parser.add_argument('destination_blob_name')
+Creating mocks for external resources is optional, and not recommended in
+most cases.
 
-    args = parser.parse_args()
+## Temporary Resources
 
-    if args.command == 'list':
-        list_blobs(args.bucket_name)
-    elif args.command == 'upload':
-        upload_blob(
-            args.bucket_name,
-            args.source_file_name,
-            args.destination_blob_name)
-```
+When tests need temporary resources (such as a temp file or folder), they
+should create reasonable names for these resources with a UUID attached to
+assure uniqueness. All temporary resources should be explicitly deleted when
+testing is complete. 
 
-### Local server
+## Console Output
 
-For web application samples using Flask that don't run on App Engine Standard,
-the `if __name__ == '__main__'` clause should handle starting the development
-server:
+If the sample prints output to the console, the test should capture stdout to
+a file and verify that the captured output contains the key information that
+is expected. Strive to verify the content of the output rather than the syntax.
+For example, the test might verify that a string is included in the output,
+without taking a dependency on where that string occurs in the output.
 
-```python
-if __name__ == '__main__':
-    # This is used when running locally. Gunicorn is used to run the
-    # application on Google App Engine. See entrypoint in app.yaml.
-    app.run(host='127.0.0.1', port=8080, debug=True)
-```
-## Writing tests
+## Using nox
```

Comment:
This section may change a little bit if we integrate @kurtisvg 's propsed noxfile changes #2806